server/tilesource/tiff.py

```diff
@@ -66,6 +67,19 @@ def __init__(self, item, **kwargs):
             try:
                 td = TiledTiffDirectory(largeImagePath, directoryNum)
             except ValidationTiffException as lastException:
+                try:
+                    associated = TiledTiffDirectory(largeImagePath, directoryNum, False)
+                    id = associated._tiffInfo.get(
+                        'imagedescription').strip().split(None, 1)[0].lower()
+                    # Only use this as an associated image if the parsed id is
+                    # a reasonable length, alphanumeric characters, and the
+                    # image isn't too large.
+                    if (id.isalnum() and len(id) > 3 and len(id) <= 20 and
+                            associated._pixelInfo['width'] <= 8192 and
+                            associated._pixelInfo['height'] <= 8192):
+                        self._associatedImages[id] = associated._tiffFile.read_image()
+                except Exception:
```

Comment:
Is it possible to use a more specific exception?  Also, the complexity of this function is getting a little high.  How about breaking this block into a new function?